### PR TITLE
[IMP] account_edi: improve how we handle attachments for the email

### PR DESCRIPTION
**Pull request by jco-odoo**
When assembling the email we are leaving the assignment of the attachments
to another method that we can inherit and then modify instead of doing it
in the same method to attach them.
The idea is that in any localization we need only to modify that method
in order to add new attachments to the mail template.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
Replacement of https://github.com/odoo/odoo/pull/69576/files

**Comment by jco-odoo** [comment]
Replacement of https://github.com/odoo/enterprise/pull/17840

File: addons/account_edi/models/mail_template.py [hunk 1]
```diff
@@ -26,10 +31,7 @@ def generate_email(self, res_ids, fields):
                 # wizard.
                 if doc.edi_format_id._is_embedding_to_invoice_pdf_needed():
                     continue
-
-                attachment = doc.attachment_id
-                if attachment:
-                    record_data.setdefault('attachments', [])
-                    record_data['attachments'].append((attachment.name, attachment.datas))
+                record_data.setdefault('attachments', [])
+                record_data['attachments'] += self._get_edi_attachments(doc)
```
Comment (smetl) [inline addons/account_edi/models/mail_template.py:35]: why don't you call this method only if the document has an attachment?

File: addons/account_edi/models/mail_template.py [hunk 2]
```diff
@@ -6,6 +6,11 @@
 class MailTemplate(models.Model):
     _inherit = "mail.template"
 
+    def _get_edi_attachments(self, document):
+        if not document.attachment_id:
```
Comment (smetl) [inline addons/account_edi/models/mail_template.py:10]: need docstring

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**